financial-connections/src/main/java/com/stripe/android/financialconnections/core/PaneViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.stripe.android.financialconnections.core
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+internal abstract class PaneViewModel<S>(
+    initialState: S
+) : ViewModel() {
+
+    val stateFlow: MutableStateFlow<S> = MutableStateFlow(initialState)
+
+    protected open fun <T : Any?> (suspend () -> T).execute(
+        reducer: S.(Result<T>) -> S,
+        onSuccess: (T) -> Unit = {},
+        onFail: (Throwable) -> Unit = {}
```

Comment:
Added some side effect callbacks that should replace the `onAsync` behavior (IMO code is easier to follow if side effects follow the operation execution)